syntax = "proto3";

option csharp_namespace = "ConduitRawAPI";

package greet;

// The greeting service definition.
service ConduitRaw {
  // Sends a greeting
	rpc Ping (PingRequest) returns (PingResponse);
	rpc GetBlockNumber (BlockNumberRequest) returns (BlockNumberResponse);
	rpc GetBlock (BlockRequest) returns (BlockResponse);
	rpc GetMerkleTreeRow (MerkleTreeRowRequest) returns (MerkleTreeRowResponse);
	rpc GetTransactionOffsets (TransactionOffsetsRequest) returns (TransactionOffsetsResponse);
	rpc GetBlockMetadata (BlockMetadataRequest) returns (BlockMetadataResponse);
}

// The request message containing the user's name.
message PingRequest {
	string data = 1;
}

// The response message containing the greetings.
message PingResponse {
	string message = 1;
}

message BlockNumberRequest {
	bytes blockHash = 1;
}

message BlockNumberResponse {
	uint64 blockNumber = 1;
}

message BlockRequest {
	uint32 blockNumber = 1;
}

message BlockResponse {
	bytes rawBlock = 1;
}

message MerkleTreeRowRequest {
	bytes blockHash = 1;
	uint32 level = 2;
}

message MerkleTreeRowResponse {
	bytes mtreeRow = 1;
}

message TransactionOffsetsRequest {
	bytes blockHash = 1;
}

message TransactionOffsetsResponse {
	repeated uint64 txOffsetsArray = 1;
}

message BlockMetadataRequest {
	bytes blockHash = 1;
}

message BlockMetadataResponse {
	uint64 blockSizeBytes = 1;
}
